From fc87e7869a9c15f393fd964f0e18829c20891f00 Mon Sep 17 00:00:00 2001
From: Even Xu <even.xu@intel.com>
Date: Thu, 6 Feb 2020 14:39:16 +0800
Subject: [PATCH 1/2] hid: intel-ish-hid: ipc: finish power flow for EHL OOB

1. Add pci_set_power_state() to clean PME assert when resume from Sx.
2. Add device_init_wakeup() to enable wake capability.

Signed-off-by: Even Xu <even.xu@intel.com>
Signed-off-by: Chinnu, Ganapathi <ganapathi.chinnu@intel.com>
---
 drivers/hid/intel-ish-hid/ipc/pci-ish.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/hid/intel-ish-hid/ipc/pci-ish.c b/drivers/hid/intel-ish-hid/ipc/pci-ish.c
index f491d8b..1d58194 100644
--- a/drivers/hid/intel-ish-hid/ipc/pci-ish.c
+++ b/drivers/hid/intel-ish-hid/ipc/pci-ish.c
@@ -174,6 +174,12 @@ static int ish_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 	init_waitqueue_head(&ishtp->suspend_wait);
 	init_waitqueue_head(&ishtp->resume_wait);
 
+	/* Enable PME for EHL */
+	if (pdev->device == EHL_Ax_DEVICE_ID) {
+		device_init_wakeup(&pdev->dev, 1);
+		dev_warn(dev, "evenadd ehl wakeup capability\n");
+	}
+
 	ret = ish_init(ishtp);
 	if (ret)
 		return ret;
@@ -311,6 +317,12 @@ static int __maybe_unused ish_resume(struct device *device)
 	struct pci_dev *pdev = to_pci_dev(device);
 	struct ishtp_device *dev = pci_get_drvdata(pdev);
 
+	/* add this to finish power flow for EHL */
+	if (dev->pdev->device == EHL_Ax_DEVICE_ID) {
+		pci_set_power_state(pdev, PCI_D0);
+		dev_warn(dev->devc, "set power state to D0 for ehl\n");
+	}
+
 	ish_resume_device = device;
 	dev->resume_flag = 1;
 
-- 
2.7.4

